mlir/lib/Dialect/Rock/Pipelines/XMIRPipelines.cpp

```diff
@@ -38,28 +38,40 @@ void xmir::buildModelPipeline(OpPassManager &pm,
   pm.addPass(rock::createRockApplyImplPass());
 }
 
+// Runner takes an Affine/SCF program with async retargetable launchs
+// and lowers to host LLVM runtime program. JitRunner then calls ORC
+// to generate X86 binary and runs it.
 void xmir::buildRunnerPipeline(OpPassManager &pm,
                                const xmir::RunnerOptions &options) {
   pm.addNestedPass<func::FuncOp>(createConvertLinalgToAffineLoopsPass());
   pm.addPass(createLowerAffinePass());
-  pm.addPass(createConvertSCFToCFPass());
-  if (options.cpuOnly) {
+  pm.addNestedPass<func::FuncOp>(createConvertSCFToCFPass());
+
+  // Target async.launch to cpu.coro or gpu.launch_func
+  pm.addPass(createConvertAsyncToGPUPass());
+  pm.addPass(createAsyncParallelForPass());
+  pm.addPass(createAsyncToAsyncRuntimePass(options.enableCoroutines));
```

Comment:
@krzysz00 if you have some time to look at coroutines, you can run xmir/e2e/resnet50_blk.cpu.mlir test but you would need set this parameter to true.